### DISCUSSION TITLE
[Enhancement] [segmented] The "el-segmented" control cannot return to the unselected state through binding values after an option is selected.

### DISCUSSION THREAD

**Discussion by haifenggo** [element-plus/element-plus#18803] [Feature Request]
Feature type: **`Enhancement`**

## Details

- Enhanced component name: `segmented`

## Description

The "el-segmented" control can default to having no option selected, but once an option is selected, it cannot be returned to the unselected state; that is to say, the highlight cannot be removed. Setting the model-value/v-model binding to null or undefined does not work; only the text color changes from white to black when selected, but the default background remains blue.

el-segmented默认可以未选中任何一个选项，但是一旦选中一个选项后，无法通过来恢复到未选择状态，也就是说无法去除高亮状态。model-value/v-model绑定值设为null和undifined都不行，只有字体从选中状态的白色转为黑色，但默认背景仍是蓝色。

### Additional comments

_(empty)_

**Comment by DDDDD12138** [comment]
没有吧，我看着是正常的: [segmented demo](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZGl2IGNsYXNzPVwiZmxleCBmbGV4LWNvbCBpdGVtcy1zdGFydCBnYXAtNFwiPlxuICAgIDxlbC1zZWdtZW50ZWQgdi1tb2RlbD1cInZhbHVlXCIgOm9wdGlvbnM9XCJvcHRpb25zXCIgLz5cbiAgPC9kaXY+XG4gIDxFbEJ1dHRvbiBAY2xpY2s9XCJ2YWx1ZSA9IG51bGxcIj7muIXnqbp2YWx1ZTwvRWxCdXR0b24+XG48L3RlbXBsYXRlPlxuPHNjcmlwdCBsYW5nPVwidHNcIiBzZXR1cD5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcblxuY29uc3QgdmFsdWUgPSByZWYoJ01vbicpXG5cbmNvbnN0IG9wdGlvbnMgPSBbJ01vbicsICdUdWUnLCAnV2VkJywgJ1RodScsICdGcmknLCAnU2F0JywgJ1N1biddXG48L3NjcmlwdD5cbiIsImVsZW1lbnQtcGx1cy5qcyI6ImltcG9ydCBFbGVtZW50UGx1cyBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBnZXRDdXJyZW50SW5zdGFuY2UgfSBmcm9tICd2dWUnXG5cbmxldCBpbnN0YWxsZWQgPSBmYWxzZVxuYXdhaXQgbG9hZFN0eWxlKClcblxuZXhwb3J0IGZ1bmN0aW9uIHNldHVwRWxlbWVudFBsdXMoKSB7XG4gIGlmIChpbnN0YWxsZWQpIHJldHVyblxuICBjb25zdCBpbnN0YW5jZSA9IGdldEN1cnJlbnRJbnN0YW5jZSgpXG4gIGluc3RhbmNlLmFwcENvbnRleHQuYXBwLnVzZShFbGVtZW50UGx1cylcbiAgaW5zdGFsbGVkID0gdHJ1ZVxufVxuXG5leHBvcnQgZnVuY3Rpb24gbG9hZFN0eWxlKCkge1xuICBjb25zdCBzdHlsZXMgPSBbJ2h0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vZWxlbWVudC1wbHVzQGxhdGVzdC9kaXN0L2luZGV4LmNzcycsICdodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL2VsZW1lbnQtcGx1c0BsYXRlc3QvdGhlbWUtY2hhbGsvZGFyay9jc3MtdmFycy5jc3MnXS5tYXAoKHN0eWxlKSA9PiB7XG4gICAgcmV0dXJuIG5ldyBQcm9taXNlKChyZXNvbHZlLCByZWplY3QpID0+IHtcbiAgICAgIGNvbnN0IGxpbmsgPSBkb2N1bWVudC5jcmVhdGVFbGVtZW50KCdsaW5rJylcbiAgICAgIGxpbmsucmVsID0gJ3N0eWxlc2hlZXQnXG4gICAgICBsaW5rLmhyZWYgPSBzdHlsZVxuICAgICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdsb2FkJywgcmVzb2x2ZSlcbiAgICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignZXJyb3InLCByZWplY3QpXG4gICAgICBkb2N1bWVudC5ib2R5LmFwcGVuZChsaW5rKVxuICAgIH0pXG4gIH0pXG4gIHJldHVybiBQcm9taXNlLmFsbFNldHRsZWQoc3R5bGVzKVxufVxuIiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZSxcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsIlBsYXlncm91bmRNYWluLnZ1ZSI6IjxzY3JpcHQgc2V0dXA+XG5pbXBvcnQgQXBwIGZyb20gJy4vQXBwLnZ1ZSdcbmltcG9ydCB7IHNldHVwRWxlbWVudFBsdXMgfSBmcm9tICcuL2VsZW1lbnQtcGx1cy5qcydcbnNldHVwRWxlbWVudFBsdXMoKVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPEFwcCAvPlxuPC90ZW1wbGF0ZT5cbiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHtcbiAgICBcInZ1ZVwiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vQHZ1ZS9ydW50aW1lLWRvbUBsYXRlc3QvZGlzdC9ydW50aW1lLWRvbS5lc20tYnJvd3Nlci5qc1wiLFxuICAgIFwiQHZ1ZS9zaGFyZWRcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0B2dWUvc2hhcmVkQGxhdGVzdC9kaXN0L3NoYXJlZC5lc20tYnVuZGxlci5qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9lbGVtZW50LXBsdXNAbGF0ZXN0L2Rpc3QvaW5kZXguZnVsbC5taW4ubWpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXMvXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9lbGVtZW50LXBsdXNAbGF0ZXN0L1wiLFxuICAgIFwiQGVsZW1lbnQtcGx1cy9pY29ucy12dWVcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlQDIvZGlzdC9pbmRleC5taW4uanNcIlxuICB9LFxuICBcInNjb3Blc1wiOiB7fVxufSIsIl9vIjp7fX0=)

**Reply by haifenggo** [reply]
我在你提供的Playground那里写了我的不能正常消除高亮的demo。

demo是有两个segmented，两个都点高亮后，然后通过watch来进行清除另一个高亮，但是只有字体从选中状态的白色转为黑色，背景仍是蓝色。

```
<template>
  <div class="flex flex-col items-start gap-4">
    <el-segmented v-model="value" :options="options" />
  </div>
  <ElButton @click="value = -1">清空value</ElButton>
       <el-segmented class="common-java" v-model="activeCommonCode" :options="CommonCodeOptions"
                      direction="vertical" :block="false"
                      size="small"
                      @change="changeCommonCode"/>
      <el-segmented class="frontend-code" v-model="activeFrontendCode" :options="frontendCodeOptions"
                      direction="vertical" :block="false"
                      size="small"
                      @change="changeFrontendCode"/>
</template>
<script lang="ts" setup>
import { ref, watch } from 'vue'
// *******************************************************
const commonCode = ref(null);
const activeCommonCode = ref('')
const CommonCodeOptions = ref([
  'a',
  'b',
  'v',
  'd',
  'e',
  'f',
  'g'
])
async function changeCommonCode(val) {
  activeCommonCode.value = val
}
// *******************************************************
const frontendCode = ref({})
const activeFrontendCode = ref('')
const frontendCodeOptions = ref([
    '1',
    '2',
    '3',
])
async function changeFrontendCode(val){
  activeFrontendCode.value = val
}
// *******************************************************
watch(activeFrontendCode, (val) => {
  if (frontendCodeOptions.value.includes(val)) {
    activeCommonCode.value = null
  }
});
// *******************************************************
const value = ref('')

const options = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun']
</script>

```

**Comment by DDDDD12138** [comment]
确实是个 BUG ，你去提个 issues 吧

**Comment by DDDDD12138** [comment]
看看这样是不是可以了呢： [demo](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZGl2IGNsYXNzPVwiZmxleCBmbGV4LWNvbCBpdGVtcy1zdGFydCBnYXAtNFwiPlxuICAgIDxlbC1zZWdtZW50ZWQgdi1tb2RlbD1cInZhbHVlXCIgOm9wdGlvbnM9XCJvcHRpb25zXCIgLz5cbiAgPC9kaXY+XG4gIDxFbEJ1dHRvbiBAY2xpY2s9XCJ2YWx1ZSA9IC0xXCI+5riF56m6dmFsdWU8L0VsQnV0dG9uPlxuICAgICAgIDxlbC1zZWdtZW50ZWQgY2xhc3M9XCJjb21tb24tamF2YVwiIHYtbW9kZWw9XCJhY3RpdmVDb21tb25Db2RlXCIgOm9wdGlvbnM9XCJDb21tb25Db2RlT3B0aW9uc1wiXG4gICAgICAgICAgICAgICAgICAgICAgZGlyZWN0aW9uPVwidmVydGljYWxcIiA6YmxvY2s9XCJmYWxzZVwiXG4gICAgICAgICAgICAgICAgICAgICAgc2l6ZT1cInNtYWxsXCJcbiAgICAgICAgICAgICAgICAgICAgICBAY2hhbmdlPVwiY2hhbmdlQ29tbW9uQ29kZVwiLz5cbiAgICAgIDxlbC1zZWdtZW50ZWQgY2xhc3M9XCJmcm9udGVuZC1jb2RlXCIgdi1tb2RlbD1cImFjdGl2ZUZyb250ZW5kQ29kZVwiIDpvcHRpb25zPVwiZnJvbnRlbmRDb2RlT3B0aW9uc1wiXG4gICAgICAgICAgICAgICAgICAgICAgZGlyZWN0aW9uPVwidmVydGljYWxcIiA6YmxvY2s9XCJmYWxzZVwiXG4gICAgICAgICAgICAgICAgICAgICAgc2l6ZT1cInNtYWxsXCJcbiAgICAgICAgICAgICAgICAgICAgICBAY2hhbmdlPVwiY2hhbmdlRnJvbnRlbmRDb2RlXCIvPlxuPC90ZW1wbGF0ZT5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyByZWYsIHdhdGNoIH0gZnJvbSAndnVlJ1xuLy8gKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKlxuY29uc3QgY29tbW9uQ29kZSA9IHJlZihudWxsKTtcbmNvbnN0IGFjdGl2ZUNvbW1vbkNvZGUgPSByZWYoJycpXG5jb25zdCBDb21tb25Db2RlT3B0aW9ucyA9IHJlZihbXG4gICdhJyxcbiAgJ2InLFxuICAndicsXG4gICdkJyxcbiAgJ2UnLFxuICAnZicsXG4gICdnJ1xuXSlcbmFzeW5jIGZ1bmN0aW9uIGNoYW5nZUNvbW1vbkNvZGUodmFsKSB7XG4gIGFjdGl2ZUNvbW1vbkNvZGUudmFsdWUgPSB2YWxcbn1cbi8vICoqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKipcbmNvbnN0IGZyb250ZW5kQ29kZSA9IHJlZih7fSlcbmNvbnN0IGFjdGl2ZUZyb250ZW5kQ29kZSA9IHJlZignJylcbmNvbnN0IGZyb250ZW5kQ29kZU9wdGlvbnMgPSByZWYoW1xuICAgICcxJyxcbiAgICAnMicsXG4gICAgJzMnLFxuXSlcbmFzeW5jIGZ1bmN0aW9uIGNoYW5nZUZyb250ZW5kQ29kZSh2YWwpe1xuICBhY3RpdmVGcm9udGVuZENvZGUudmFsdWUgPSB2YWxcbn1cbi8vICoqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKipcbndhdGNoKGFjdGl2ZUZyb250ZW5kQ29kZSwgKHZhbCkgPT4ge1xuICBpZiAoZnJvbnRlbmRDb2RlT3B0aW9ucy52YWx1ZS5pbmNsdWRlcyh2YWwpKSB7XG4gICAgYWN0aXZlQ29tbW9uQ29kZS52YWx1ZSA9IG51bGxcbiAgfVxufSk7XG4vLyAqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqKioqXG5jb25zdCB2YWx1ZSA9IHJlZignJylcblxuY29uc3Qgb3B0aW9ucyA9IFsnTW9uJywgJ1R1ZScsICdXZWQnLCAnVGh1JywgJ0ZyaScsICdTYXQnLCAnU3VuJ11cbjwvc2NyaXB0PlxuIiwiZWxlbWVudC1wbHVzLmpzIjoiaW1wb3J0IEVsZW1lbnRQbHVzIGZyb20gJ2VsZW1lbnQtcGx1cydcbmltcG9ydCB7IGdldEN1cnJlbnRJbnN0YW5jZSB9IGZyb20gJ3Z1ZSdcblxubGV0IGluc3RhbGxlZCA9IGZhbHNlXG5hd2FpdCBsb2FkU3R5bGUoKVxuXG5leHBvcnQgZnVuY3Rpb24gc2V0dXBFbGVtZW50UGx1cygpIHtcbiAgaWYgKGluc3RhbGxlZCkgcmV0dXJuXG4gIGNvbnN0IGluc3RhbmNlID0gZ2V0Q3VycmVudEluc3RhbmNlKClcbiAgaW5zdGFuY2UuYXBwQ29udGV4dC5hcHAudXNlKEVsZW1lbnRQbHVzKVxuICBpbnN0YWxsZWQgPSB0cnVlXG59XG5cbmV4cG9ydCBmdW5jdGlvbiBsb2FkU3R5bGUoKSB7XG4gIGNvbnN0IHN0eWxlcyA9IFsnaHR0cHM6Ly9wcmV2aWV3LTE4ODE4LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzJywgJ2h0dHBzOi8vcHJldmlldy0xODgxOC1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmNzcyddLm1hcCgoc3R5bGUpID0+IHtcbiAgICByZXR1cm4gbmV3IFByb21pc2UoKHJlc29sdmUsIHJlamVjdCkgPT4ge1xuICAgICAgY29uc3QgbGluayA9IGRvY3VtZW50LmNyZWF0ZUVsZW1lbnQoJ2xpbmsnKVxuICAgICAgbGluay5yZWwgPSAnc3R5bGVzaGVldCdcbiAgICAgIGxpbmsuaHJlZiA9IHN0eWxlXG4gICAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2xvYWQnLCByZXNvbHZlKVxuICAgICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdlcnJvcicsIHJlamVjdClcbiAgICAgIGRvY3VtZW50LmJvZHkuYXBwZW5kKGxpbmspXG4gICAgfSlcbiAgfSlcbiAgcmV0dXJuIFByb21pc2UuYWxsU2V0dGxlZChzdHlsZXMpXG59XG4iLCJ0c2NvbmZpZy5qc29uIjoie1xuICBcImNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogXCJFU05leHRcIixcbiAgICBcImpzeFwiOiBcInByZXNlcnZlXCIsXG4gICAgXCJtb2R1bGVcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVJlc29sdXRpb25cIjogXCJCdW5kbGVyXCIsXG4gICAgXCJ0eXBlc1wiOiBbXCJlbGVtZW50LXBsdXMvZ2xvYmFsLmQudHNcIl0sXG4gICAgXCJhbGxvd0ltcG9ydGluZ1RzRXh0ZW5zaW9uc1wiOiB0cnVlLFxuICAgIFwiYWxsb3dKc1wiOiB0cnVlLFxuICAgIFwiY2hlY2tKc1wiOiB0cnVlXG4gIH0sXG4gIFwidnVlQ29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiAzLjNcbiAgfVxufVxuIiwiUGxheWdyb3VuZE1haW4udnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCBBcHAgZnJvbSAnLi9BcHAudnVlJ1xuaW1wb3J0IHsgc2V0dXBFbGVtZW50UGx1cyB9IGZyb20gJy4vZWxlbWVudC1wbHVzLmpzJ1xuc2V0dXBFbGVtZW50UGx1cygpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8QXBwIC8+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwidnVlXCI6IFwiaHR0cHM6Ly9jZG4uanNkZWxpdnIubmV0L25wbS9AdnVlL3J1bnRpbWUtZG9tQGxhdGVzdC9kaXN0L3J1bnRpbWUtZG9tLmVzbS1icm93c2VyLmpzXCIsXG4gICAgXCJAdnVlL3NoYXJlZFwiOiBcImh0dHBzOi8vY2RuLmpzZGVsaXZyLm5ldC9ucG0vQHZ1ZS9zaGFyZWRAbGF0ZXN0L2Rpc3Qvc2hhcmVkLmVzbS1idW5kbGVyLmpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXNcIjogXCJodHRwczovL3ByZXZpZXctMTg4MTgtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5mdWxsLm1pbi5tanNcIixcbiAgICBcImVsZW1lbnQtcGx1cy9cIjogXCJ1bnN1cHBvcnRlZFwiLFxuICAgIFwiQGVsZW1lbnQtcGx1cy9pY29ucy12dWVcIjogXCJodHRwczovL2Nkbi5qc2RlbGl2ci5uZXQvbnBtL0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlQDIvZGlzdC9pbmRleC5taW4uanNcIlxuICB9LFxuICBcInNjb3Blc1wiOiB7fVxufSIsIl9vIjp7InNob3dIaWRkZW4iOnRydWUsInN0eWxlU291cmNlIjoiaHR0cHM6Ly9wcmV2aWV3LTE4ODE4LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzIn19)

**Comment by thinkasany** [comment]
确实有问题，#18818 这个pr 会处理这个问题
